### ISSUE TITLE
the value of the loss function appears to be nan

### ISSUE THREAD

**Issue by CurryxIaoHu** [franciscoliu/SKU#1]
Hi! Thanks for the amazing work!
When I reproduced the code, the value of the loss function appears to be nan during the finetuning process: 

```
batch: 0, GD_loss: 8.13, RD_loss: 7.95, reversed_kl_loss: nan, combined_loss: nan, 
batch: 1, GD_loss: nan, RD_loss: nan, reversed_kl_loss: nan, combined_loss: nan, 
batch: 2, GD_loss: nan, RD_loss: nan, reversed_kl_loss: nan, combined_loss: nan, 
batch: 3, GD_loss: nan, RD_loss: nan, reversed_kl_loss: nan, combined_loss: nan, 
batch: 4, GD_loss: nan, RD_loss: nan, reversed_kl_loss: nan, combined_loss: nan, 
batch: 5, GD_loss: nan, RD_loss: nan, reversed_kl_loss: nan, combined_loss: nan, 
batch: 6, GD_loss: nan, RD_loss: nan, reversed_kl_loss: nan, combined_loss: nan, 
batch: 7, GD_loss: nan, RD_loss: nan, reversed_kl_loss: nan, combined_loss: nan, 
batch: 8, GD_loss: nan, RD_loss: nan, reversed_kl_loss: nan, combined_loss: nan, 
batch: 9, GD_loss: nan, RD_loss: nan, reversed_kl_loss: nan, combined_loss: nan, 
```

I did not change the hyperparameters provided in the demo except for the learning rate. I have tried to increase or decrease the learning rate, but none of it helped.

**Comment by franciscoliu** [comment]
Hi, thank you for the question. It is really hard to tell from the output you are given. I have reran the code on my side and it did not appear the problem. Have you tried to comment out each loss and see if they works fine individually?

**Comment by AnonymousUserA** [comment]
Indeed, the gradient explodes around the sixth iteration, and the loss is NAN.

**Comment by franciscoliu** [comment]
I think the problem may occur at the accumulating the different loss functions together, could you try to assign lower weights to reversed_kl_loss and more weight to RD_loss and see if the problem resolves?

**Comment by AnonymousUserA** [comment]
Thanks! I solved the exploding gradient problem by gradient clipping, I was wondering if you could share your fine-tuned model to me?
